library/core/src/num/niche_types.rs

```diff
@@ -14,7 +14,8 @@ macro_rules! define_valid_range_type {
         $(#[$m:meta])*
         $vis:vis struct $name:ident($int:ident as $uint:ident in $low:literal..=$high:literal);
     )+) => {$(
-        #[derive(Clone, Copy, Eq)]
+        #[derive(Copy)]
+        #[derive_const(Clone, Eq)]
```

Comment:
Let's do that after the pattern types refactor lands. Otherwise I may just have to temporarily remove it again